src/data/client.js

```diff
@@ -12,8 +12,10 @@ export const getDataTypes = async (callback) => {
   return callback(data);
 }
 
-export const getDatasets = async (callback, query, categories) => {
-  const response = await fetch(`api/datasets/search?query=${query}&categories=${categories}`)
+export const getDatasets = async (callback, query, categories, dataTypes) => {
+  const cat = categories == "\"\""? "" : categories
+  const dat = (dataTypes == "\"\"" || dataTypes == undefined)? "" : dataTypes
```

Comment:
Can you add comments explaining what these checks are for? Anyone who comes back and looks at this will have no idea what's going on.

Reply:
I moved the conditional to Datasets.js and switched it to checking the length before stringifying.
It's way more intuitive so no comments needed. 
Ah, and the undefined was due to me forgetting to add dataType field for one of the getDatasets calls.